pyocd/subcommands/rtt_cmd.py

```diff
@@ -120,27 +124,48 @@ def invoke(self) -> int:
                 memory_map: MemoryMap = target.get_memory_map()
                 ram_region: MemoryRegion = memory_map.get_default_region_of_type(MemoryType.RAM)
 
-                if self._args.address is None or self._args.size is None:
-                    rtt_range_start = ram_region.start
-                    rtt_range_size = ram_region.length
-                elif ram_region.start <= self._args.address and self._args.size <= ram_region.length:
-                    rtt_range_start = self._args.address
-                    rtt_range_size = self._args.size
-
-                LOG.info(f"RTT control block search range [{rtt_range_start:#08x}, {rtt_range_size:#08x}]")
-
                 rtt_cb_addr = -1
-                data = bytearray(b'0000000000')
-                chunk_size = 1024
-                while rtt_range_size > 0:
-                    read_size = min(chunk_size, rtt_range_size)
-                    data += bytearray(target.read_memory_block8(rtt_range_start, read_size))
-                    pos = data[-(read_size + 10):].find(b"SEGGER RTT")
-                    if pos != -1:
-                        rtt_cb_addr = rtt_range_start + pos - 10
-                        break
-                    rtt_range_start += read_size
-                    rtt_range_size -= read_size
+
+                elf_file_arg, elf_file_option = self._args.elf, session.options.get("rtt.elf")
+                if elf_file_arg != None:
+                    elf_file = elf_file_arg
+                elif elf_file_option != None:
+                    elf_file = elf_file_option
+                else:
+                    elf_file = None
+
+                if elf_file != None:
+                    elf = ELFFile(open(self._args.elf, "rb"))
+                    symtab = elf.get_section_by_name(".symtab")
+                    if symtab:
+                        rtt_entry = symtab.get_symbol_by_name("_SEGGER_RTT")
+                        if len(rtt_entry) > 0:
+                            rtt_cb_addr = rtt_entry[0].entry["st_value"]
```

Comment:
Fyi, pyocd has some built-in support for ELF symbol extraction (and other things) in the `pyocd.debug.elf` sub-package. It's based on pyelftools, so for cases like this, where there is not much advantage, it's fine to directly use pyelftools.

For reference, this is what this code would look like using pyocd's ELF tools:

```python
sym_info = ELFBinaryFile(self._args.elf).symbol_decoder.get_symbol_by_name('_SEGGER_RTT')
if sym_info:
    rtt_cb_addr = sym_info.value
```